src/bridge/connectors/connector_6.rs

```diff
@@ -0,0 +1,93 @@
+use std::collections::HashMap;
+
+use crate::{
+    bridge::{
+        constants::{
+            DESTINATION_NETWORK_TXID_LENGTH_IN_DIGITS, SOURCE_NETWORK_TXID_LENGTH_IN_DIGITS,
+        },
+        graphs::peg_out::CommitmentMessageId,
+        transactions::{base::Input, signing_winternitz::WinternitzPublicKeyVariant},
+    },
+    signatures::{winternitz::PublicKey, winternitz_hash::check_hash_sig},
+    treepp::script,
+};
+use bitcoin::{
+    key::Secp256k1,
+    taproot::{TaprootBuilder, TaprootSpendInfo},
+    Address, Network, ScriptBuf, TxIn, XOnlyPublicKey,
+};
+
+use serde::{Deserialize, Serialize};
+
+use super::base::{generate_default_tx_in, TaprootConnector};
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Clone)]
+pub struct Connector6 {
+    pub network: Network,
+    pub operator_taproot_public_key: XOnlyPublicKey,
+    pub commitment_public_keys: HashMap<CommitmentMessageId, WinternitzPublicKeyVariant>,
+}
+
+impl Connector6 {
+    pub fn new(
+        network: Network,
+        operator_taproot_public_key: &XOnlyPublicKey,
+        commitment_public_keys: &HashMap<CommitmentMessageId, WinternitzPublicKeyVariant>,
+    ) -> Self {
+        Connector6 {
+            network,
+            operator_taproot_public_key: operator_taproot_public_key.clone(),
+            commitment_public_keys: commitment_public_keys.clone(),
+        }
+    }
+
+    fn generate_taproot_leaf_0_script(&self) -> ScriptBuf {
+        let destination_network_txid_public_key = self.commitment_public_keys
+            [&CommitmentMessageId::PegOutTxIdDestinationNetwork]
+            .get_standard_variant_ref();
+        let source_network_txid_public_key = self.commitment_public_keys
+            [&CommitmentMessageId::PegOutTxIdSourceNetwork]
+            .get_standard_variant_ref();
+
+        script! {
+          { check_hash_sig(&PublicKey::from(destination_network_txid_public_key), DESTINATION_NETWORK_TXID_LENGTH_IN_DIGITS) }
```

Comment:
Are you sure you want to use `check_hash_sig` for the txids? If they are 32 bytes it should be cheaper to use them as the message directly instead of hashing them with the `blake3` script. If you want to use `check_hash_sig` then you would have to provide the message in bytes not digits.

Reply:
Just compared that in a test and using a Winternitz signature directly uses about 10k opcodes while only hashing 64 Bytes is already 100k.